// node_db/DbCenter.js
var mysql=require("mysql");
var logger = require('node_log').logger.getLogger();

var DbCenter = function(){};
DbCenter.prototype.query = function(sql,options,callback)
{
    var pool = mysql.createPool({
        host : '192.168.0.101',
        port : 3306,//端口号
        database : 'fengtian',//数据库名
        user : 'root',//数据库用户名
        password : 'abc.123'//数据库密码

        // host : '127.0.0.1',
        // port : 3306,//端口号
        // database : 'fengtian',//数据库名
        // user : 'root',//数据库用户名
        // password : '123456'//数据库密码
    });
    pool.getConnection(function(err,conn){
        if(err){
            logger.info("----------------数据库连接失败！--------------------");
            logger.info(err);
            callback(err,null,null);
        }else{
            conn.query(sql,options,function(err,results,fields){
                //事件驱动回调
                callback(err,results,fields);
            });
            //释放连接，需要注意的是连接释放需要在此处释放，而不是在查询回调里面释放
            conn.release();
        }
    });


};
module.exports = new DbCenter();
