// node_log/Logger.js
const log4js = require('log4js');
log4js.configure({
    replaceConsole: true,
    appenders: {
        stdout: {//控制台输出
            type: 'console'
        },
        info: {
            type: 'dateFile',
            filename: 'logs/infolog/',
            pattern: 'info-yyyy-MM-dd.log',
            alwaysIncludePattern: true
        },
        // error: {
        //     type: 'dateFile',
        //     filename: 'logs/errorlog/',
        //     pattern: 'error-yyyy-MM-dd.log',
        //     alwaysIncludePattern: true
        // },
    },
    categories: {
        default: {appenders: ['stdout', 'info'], level: 'info'},
        // error: {appenders: ['stdout', 'error'], level: 'error'},
    }
});
var Logger = function () {
};
Logger.prototype.getLogger = function (name) {//name取categories项
    return log4js.getLogger(name || 'info');
};
Logger.prototype.useLogger = function (app, logger) {//用来与express结合
    app.use(log4js.connectLogger(logger || log4js.getLogger('info'), {
        format: '[:remote-addr :method :url :status :response-timems][:referrer HTTP/:http-version :user-agent]'//自定义输出格式
    }))
}

module.exports = new Logger();
